packages/jobs/lib/crons/deleteSyncsData.ts

```diff
@@ -26,30 +37,40 @@ export async function deleteSyncsData(): Promise<void> {
 export async function exec(): Promise<void> {
     logger.info('[deleteSyncs] starting');
 
-    // -----
-    // Soft delete jobs
-    let countJobs = 0;
-    do {
-        countJobs = await softDeleteJobs(limitJobs);
-        logger.info(`[deleteSyncs] soft deleted ${countJobs} jobs`);
-    } while (countJobs >= limitJobs);
-
-    // -----
-    // Soft delete schedules
-    let countSchedules = 0;
-    do {
-        countSchedules = await softDeleteSchedules(limitSchedules);
-        logger.info(`[deleteSyncs] soft deleted ${countSchedules} schedules`);
-    } while (countSchedules >= limitSchedules);
-
-    // ----
-    // hard delete records
-    const syncs = await syncDataService.findSyncsWithDeletableRecords(limitSyncs);
-    logger.info(`[deleteSyncs] found ${syncs.length} syncs for records`);
-    for (const sync of syncs) {
-        logger.info(`[oldActivity] deleting syncId: ${sync.id}`);
-        await syncDataService.deleteRecordsBySyncId(sync.id!, limitRecords);
-    }
+    await db.knex.transaction(async (trx) => {
+        // Because it's slow and create deadlocks
+        // we need to acquire a Lock that prevents any other duplicate cron to execute the same thing
+        const { rows } = await trx.raw(`SELECT pg_try_advisory_xact_lock(?);`, [123456789]);
+        if (!rows || rows.lengt <= 0 || rows[0].pg_try_advisory_xact_lock === false) {
```

Comment:
```suggestion
        if (!rows || rows.length <= 0 || rows[0].pg_try_advisory_xact_lock === false) {
```

Reply:
ah good catch, that's what you get with `any` 😠 thanks